src/notation/internal/engravingfontscontroller.cpp

```diff
@@ -0,0 +1,193 @@
+    globalFontsPaths.append(QString::fromLocal8Bit(qgetenv("XDG_DATA_DIRS")).split(':'));
+#endif
+
+    // The first location is the user-wide location, so we should iterate in reverse order so that
+    // user-specific fonts can override system-wide fonts
+    for (auto it = globalFontsPaths.rbegin(); it != globalFontsPaths.rend(); ++it) {
+        scanDirectory(*it + "/SMuFL/Fonts", false);
+    }
+
+    // Scan MuseScore-specific "private" location last so that it can override global fonts
+    scanDirectory(configuration()->userMusicFontsPath(), true);
+
+    QStringList musicFonts;
+
+    for (const auto& font : engravingFonts()->fonts()) {
+        musicFonts << QString::fromStdString(font->name());
+        musicFonts << QString::fromStdString(font->name() + " Text");
+    }
+
+    uiConfiguration()->setNonTextFonts(musicFonts);
+}
+
+void EngravingFontsController::scanDirectory(const muse::io::path_t& path, bool isPrivate) const
+{
+    using namespace muse::draw;
+
+    std::shared_ptr<IFontsDatabase> fdb = fontsDatabase();
+
+    QDirIterator iterator(path.toQString(), QDir::Dirs | QDir::NoDotAndDotDot | QDir::Readable);
+
+    while (iterator.hasNext()) {
+        iterator.next();
+        QString fontDir = iterator.filePath();
+        muse::io::path_t metadataPath;
+        QDirIterator fontFiles(fontDir, { "*.json" }, QDir::Files);
+
+        while (fontFiles.hasNext()) {
+            fontFiles.next();
+            metadataPath = fontFiles.filePath();
+        }
+
+        if (metadataPath.empty()) {
+            continue;
+        }
+
+        // We assume the font name is the same as the directory name,
+        // but maybe we should instead read the metadata.json file to get the font name
+
+        muse::String fontName = iterator.fileName();
+        muse::String fontFamily = fontName;
+
+        if (fontName.contains(u"Text")) {
+            LOGI() << "Skipping text music font: " << fontName;
+            continue;
+        }
+
+        auto findFontPath = [this, isPrivate, fontDir](muse::String fontName) {
+            if (isPrivate) {
+                return findFontPathPrivate(fontDir, fontName);
+            } else {
+                return findFontPathGlobal(fontName);
+            }
+        };
+
+        muse::io::path_t symbolFontPath = findFontPath(fontName);
+        if (symbolFontPath.empty()) {
+            muse::String tmp = fontName;
+            symbolFontPath = findFontPath(tmp.replace(u" ", u""));
+        }
+
+        muse::io::path_t textFontPath = findFontPath(fontName + u" Text");
+        if (symbolFontPath.empty()) {
+            muse::String tmp = fontName;
+            symbolFontPath = findFontPath(tmp.replace(u" ", u"") + u"Text");
+        }
+
+        if (textFontPath.empty()) {
+            textFontPath = symbolFontPath;
+        }
+
+        if (symbolFontPath.empty()) {
+            LOGE() << "Music font not found for " << fontName;
```

Comment:
Could you also log the location of the corresponding metadata file here? Something like
```cpp
            LOGE() << "Music font" << fontName << "for" << metadataPath << "not found";
```

Reply:
(Happens twice in my setup, for fonts 'installed' in the gloval and user dir, Mu3 finds and loads them from there)

Reply:
Ah, having the .ttf/.otf included right next to the metadata file is only supported for private fonts (`MuseScore4/MusicFonts` dir).
But I will change that logging message to include the metadata when I do the next commit.